docs/extensions/guides/ipc.md

```diff
@@ -40,57 +40,69 @@ The general terminology is as follows:
   }
+}
+```
 
-  initializeListener = (event: Types.IpcMainEvent, id: string) => {
-    console.log(`starting to initialize: ${id}`);
-  };
+This file shows that you need to create an instance of the store to be able to use IPC.
+Lens will automatically clean up that store and all the handlers on deactivation and uninstall.
 
-  initializeVerifier = (args: unknown[]): args is [id: string] => {
-    return args.length === 1 && typeof args[0] === "string";
-  }
+---
+
+`main/helper-file.ts`:
+```typescript
+import { Store } from "@k8slens/extensions";
+
+function onInitialize(event: Types.IpcMainEvent, id: string) {
+  console.log(`starting to initialize: ${id}`);
+}
+
+export function registerListeners() {
+  Store.MainIpcStore.getInstance().listenIpc("initialize", onInitialize);
 }
 ```
 
-File2:
+In other files, it is not necessary to pass around any instances.
+It should be able to just call `getInstance()` everywhere in your extension as needed.
+
+---
+
+`renderer/extension.ts`:
 ```typescript
-import { LensMainExtension, Interface, Types } from "@k8slens/extensions";
+import { LensRendererExtension, Interface, Types } from "@k8slens/extensions";
 
 export class ExampleExtensionRenderer extends LensRendererExtension {
   onActivate() {
-    setTimeout(() => this.sendIpc("initialize", "an-id"), 5000);
+    const ipcStore = Store.RendererIpcStore.createInstance(this);
+
+    setTimeout(() => ipcStore.broadcastIpc("initialize", "an-id"), 5000);
   }
 }
 ```
 
+It is also needed to create an instance to broadcast messages too.
+
+---
+
 As this example shows: the channel names *must* be the same.
 It should also be noted that "listeners" and "handlers" are specific to either `LensRendererExtension` and `LensMainExtension`.
 There is no behind the scenes transfer of these functions.
 
-If you want to register a "handler" you would call `this.handleIpc(...)` instead.
+If you want to register a "handler" you would call `Store.MainIpcStore.handleIpc(...)` instead.
 The cleanup of these handlers is handled by Lens itself.
 
-### Note about verification
-
-We require that extension developers provide a verification function when registering a listener or handler.
-This is done as a preventative measure to help separate issues that can happen at runtime.
-While it is possible to use the unary truth function, this is highly discouraged.
-
-The verification function should do some cursory validation on the values send along the channel.
-Your handler or listener will not be called if it fails this validation.
-Instead an error or log message will occur.
-This should help with debugging because you are notified immediately that there is a mismatch between what you are expecting and what was sent.
+`Store.RendererIpcStore.broadcastIpc(...)` and `Store.MainIpcStore.broadcastIpc(...)` sends an event to all renderer frames and to main.
+Because of this, if you have the same channel name in both `main` and `renderer` both will receive the events.
```

Comment:
Yes

Reply:
Then maybe clarify this by changing the sentence to something like, "Because of this, no matter where you broadcast from, all listeners in `main` and `renderer` will be notified."? That seems clearer to me.

Reply:
Sure sounds good.